### DISCUSSION TITLE
Allow specifying icons for buttons

### DISCUSSION THREAD

**Discussion by mturoci** [h2oai/wave#681] [Ideas] [answered]
#### Is your feature request related to a problem? Please describe

Additional icons improve general UX and visuals.

#### Describe the solution you'd like

`icon` prop on `ui.button`

**Comment by lo5** [comment, marked as answer]
What would the buttons look like visually in practice?

**Reply by mturoci** [reply]
![image](https://user-images.githubusercontent.com/64769322/112942405-5a0a1500-9130-11eb-8f5a-65924aeb84f0.png)
from https://developer.microsoft.com/en-us/fluentui#/controls/web/button

**Reply by derrick56007** [reply]
this would be very useful

**Reply by lo5** [reply]
@mturoci - I didn't realize Fluent is already capable of this. Agree - `icon` on `ui.button()` would be a no-brainer.

**Comment by mturoci** [comment]
Tracked in https://github.com/h2oai/wave/issues/747.
